--- /var/tmp/glusterfs.spec.orig	2016-05-26 15:40:50.252668947 -0500
+++ /var/tmp/glusterfs.spec	2016-05-26 15:21:44.357157085 -0500
@@ -1343,6 +1343,8 @@
     %{buildroot}%{_sharedstatedir}/glusterd/hooks/1/add-brick/pre
 %{__install} -p -m 0744 extras/hook-scripts/reset/post/*.sh   \
     %{buildroot}%{_sharedstatedir}/glusterd/hooks/1/reset/post
+%{__install} -p -m 0744 extras/hook-scripts/S56glusterd-geo-rep-create-post.sh   \
+    %{buildroot}%{_sharedstatedir}/glusterd/hooks/1/gsync-create/post
 
 
 find ./tests ./run-tests.sh -type f | cpio -pd %{buildroot}%{_prefix}/share/glusterfs
